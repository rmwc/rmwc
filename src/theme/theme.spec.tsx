import React from 'react';
import { render, screen } from '@testing-library/react';
import { Theme, ThemeProvider } from './';
import { themeOptions } from './theme-options';

describe('Theme', () => {
  test('renders', () => {
    themeOptions.map((theme, i) =>
      render(
        <Theme key={i} use={theme}>
          {theme}
        </Theme>
      )
    );
  });

  test('wraps', () => {
    render(
      <Theme use="primary" wrap>
        <div className="test-classname">Hello</div>
      </Theme>
    );
    expect(screen.getByText('Hello')).toHaveClass('test-classname');
  });

  test('can have custom classnames', () => {
    const { container } = render(
      <Theme use="onPrimary" className="my-custom-classname" />
    );
    expect(container.firstChild).toHaveClass('my-custom-classname');
  });
});

describe('ThemeProvider', () => {
  test('renders', () => {
    const { asFragment } = render(
      <ThemeProvider
        options={{ primary: 'red', secondary: '#fff', surface: '#000000' }}
      >
        <div />
      </ThemeProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('can wrap', () => {
    render(
      <ThemeProvider options={{ primary: 'red' }} wrap>
        <span>Hello</span>
      </ThemeProvider>
    );

    const { asFragment } = render(
      <ThemeProvider options={{ primary: 'red' }} wrap />
    );

    expect(screen.getByText('Hello')).toBeInTheDocument();

    expect(asFragment()).toMatchInlineSnapshot(`
      <DocumentFragment>
        <div
          class=""
          style="--mdc-theme-on-primary: rgba(255, 255, 255, 1); --mdc-theme-primary: red;"
        />
      </DocumentFragment>
    `);
  });
});
