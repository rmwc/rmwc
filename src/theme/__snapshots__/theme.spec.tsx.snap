// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider renders 1`] = `
<DocumentFragment>
  <div
    class=""
    style="--mdc-theme-on-primary: rgba(255, 255, 255, 1); --mdc-theme-on-surface: rgba(255, 255, 255, 1); --mdc-theme-on-secondary: rgba(0, 0, 0, 0.87); --mdc-theme-primary: red; --mdc-theme-secondary: #fff; --mdc-theme-surface: #000000;"
  >
    <div />
  </div>
</DocumentFragment>
`;
